### PR TITLE
Keep test matrix in sync with kafka_protocol test matrix

### PR DESCRIPTION
**Pull request by robsonpeixoto**

### CODE REVIEW

File: scripts/setup-test-env.sh [hunk 1]
```diff
@@ -15,10 +15,12 @@ case $VERSION in
     VERSION="0.11";;
   1.*)
     VERSION="1.1";;
```
Comment (zmstone) [inline scripts/setup-test-env.sh:17]: maybe add `2.4*)` here
and leave `2.*)` unchanged.

Comment (robsonpeixoto) [reply scripts/setup-test-env.sh:17]: What your opinion about keep test matrix in sync with `kafka_protocol`?

- https://github.com/kafka4beam/kafka_protocol/blob/4.0.1/scripts/setup-test-env.sh#L8-L22
- https://github.com/kafka4beam/kafka_protocol/blob/4.0.1/.github/workflows/ci.yml#L12-L27

Comment (robsonpeixoto) [reply scripts/setup-test-env.sh:17]: But I did what do you ask: https://github.com/kafka4beam/brod/pull/472/commits/ad5065d97266d9a5a1627b5c17d8c6f57721327f

Comment (zmstone) [reply scripts/setup-test-env.sh:17]: > What your opinion about keep test matrix in sync with kafka_protocol?

that would be great

Comment (robsonpeixoto) [reply scripts/setup-test-env.sh:17]: Done https://github.com/kafka4beam/brod/pull/472/commits/b0955a47408c4502256c1dd27162e4ae3cba9a6b

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -14,15 +14,15 @@ jobs:
...
+            otp-version: 23.2.3-1
             kafka-version: 2.4
           - platform: ubuntu-20.04
             lsb_release: focal
             otp-version: 22.3.4.8-1
-            kafka-version: 2.2
+            kafka-version: 1.1
           - platform: ubuntu-20.04
             lsb_release: focal
-            otp-version: 24.0.2-1
```
Comment (zmstone) [inline .github/workflows/build.yml:25]: oh, ci provided rebar3 is built on otp 22.
we should test with otp24 for both brod and kafka_protocol.

Comment (robsonpeixoto) [reply .github/workflows/build.yml:25]: Please, review it: e42b774d97dd6b28925264042a698a14a034356d

**Comment by zmstone** [comment]
thank you so much.

**Pull request merged**